examples/hybrid/orographic_gravity_wave_helper.jl

```diff
@@ -0,0 +1,244 @@
+using ClimaCoreTempestRemap
+using NCDatasets
+
+function get_topo_info(Y, topo_dir, datafile_rll, comms_ctx)
+    # model grid/space information
+    cspace = axes(Y.c)
+    hspace = cspace.horizontal_space
+
+    # generate weight file for regridding
+    weightfile = create_regrid_weight_files(
+        comms_ctx,
+        topo_dir,
+        datafile_rll,
+        hspace;
+        hd_outfile_root = "topo",
+        mono = false,
+    )
+
+    varnames = ["t11", "t12", "t21", "t22", "hmin", "hmax"]
+
+    topo_info = FieldFromNamedTuple(
+        axes(Fields.level(Y.c.ρ, 1)),
+        (;
+            t11 = FT(0),
+            t12 = FT(0),
+            t21 = FT(0),
+            t22 = FT(0),
+            hmin = FT(0),
+            hmax = FT(0),
+        ),
+    )
+
+
+    for varname in varnames
+        datafile_cgll = joinpath(topo_dir, "topo_" * varname * ".g")
+        apply_remap(datafile_cgll, datafile_rll, weightfile, [varname])
+        parent(getproperty(topo_info, Symbol(varname))) .=
+            parent(convert_g2field(datafile_cgll, weightfile, hspace, varname))
+    end
+
+    # write to hdf5
+    hdfwriter = InputOutput.HDF5Writer(joinpath(topo_dir, "topo_info.hdf5"))
+    InputOutput.write!(hdfwriter, topo_info, "topo_info")
+    Base.close(hdfwriter)
+
+    return
+end
+
+function FieldFromNamedTuple(space, nt::NamedTuple)
+    cmv(z) = nt
+    return cmv.(Fields.coordinate_field(space))
+end
+
+function save_remap_hdf5(hd_outfile_root, field, varname)
+    hdfwriter =
+        InputOutput.HDF5Writer(joinpath(REGRID_DIR, hd_outfile_root * ".hdf5"))
+
+    InputOutput.write!(hdfwriter, field, string(varname))
+    Base.close(hdfwriter)
+end
+
+function hdread_regridfile(comms_ctx, hdf5file, varname)
+    hdfreader = InputOutput.HDF5Reader(hdf5file, comms_ctx)
+    field = InputOutput.read_field(hdfreader, varname)
```

Comment:
We may be able to reduce allocs using the do loop here (e.g. like [here](https://github.com/CliMA/ClimaCoupler.jl/blob/2111169492bbd40fd8584fdb4114247acb9ec2f9/test/mpi_tests/bcreader_mpi_tests.jl#L70)), but it's probably not too crucial since this happens during init only.